source/common/tcp/async_tcp_client_impl.cc

```diff
@@ -109,28 +109,31 @@ void AsyncTcpClientImpl::reportConnectionDestroy(Network::ConnectionEvent event)
 void AsyncTcpClientImpl::onEvent(Network::ConnectionEvent event) {
   if (event == Network::ConnectionEvent::RemoteClose ||
       event == Network::ConnectionEvent::LocalClose) {
-    if (disconnected_) {
+    if (connected_) {
+      cluster_info_->trafficStats()->upstream_cx_active_.dec();
```

Comment:
This is always incremented in `connect()`, but `connected_` isn't set to true there, so I think you're leaking a count on this gauge when there's a failure to connect.

Reply:
Correct! thanks, I should have changed this with last commit. Anyways, added a test that checks this scenario (it would have failed before fixing this)